components/service/glean/src/main/java/mozilla/components/service/glean/timing/TimingManager.kt

```diff
@@ -20,13 +25,18 @@ import java.util.WeakHashMap
 internal object TimingManager {
     private val logger = Logger("glean/TimingManager")
 
+    /**
+     * A incremental, running counter that represents a timer id.
+     */
+    private var timerIdCounter: GleanTimerId = 0L
+
     /**
      * A map that stores the start times of running timers.
      *
      * A [WeakHashMap] is used so that we don't unintentionally leak memory
      * by keeping references to otherwise destroyed objects around.
      */
-    val uncommittedStartTimes = mutableMapOf<String, WeakHashMap<Any, Long>>()
+    private val uncommittedStartTimes = mutableMapOf<String, WeakHashMap<GleanTimerId, Long>>()
```

Comment:
Do we still require the `WeakHashMap` here? Would a normal hash map change anything? Can primitives (which is what GleanTimerId currently is) be garbage-collected?

Reply:
Yeah -- I think the WeakHashMap doesn't buy us anything anymore.